### PR TITLE
[test] SageMaker local test failure

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [build] | [test] | [build, test] | [ec2, ecs, eks, sagemaker]

*Description:*

- Remove sleep time to avoid socket timeout issues on fabric.
- Adding rerun markers to tests that fail due to 502 gateway error.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/sagemaker.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def install_sm_local_dependencies(framework, job_type, image, ec2_conn):
         # TF inference test fail if run as soon as instance boots, even after health check pass. rootcause:
         # sockets?/nginx startup?/?
         print("sleep 300s for tensorflow inference images to avoid socket issues")
```
Comment (TusharKanekiDey) [inline test/test_utils/sagemaker.py:188]: nit: Can u keep this commit as 500s as well.

Comment (tejaschumbalkar) [reply test/test_utils/sagemaker.py]: done.

File: test/test_utils/sagemaker.py [hunk 2]
```diff
@@ -185,8 +185,6 @@ def install_sm_local_dependencies(framework, job_type, image, ec2_conn):
     if framework == "tensorflow" and job_type == "inference":
         # TF inference test fail if run as soon as instance boots, even after health check pass. rootcause:
         # sockets?/nginx startup?/?
-        print("sleep 300s for tensorflow inference images to avoid socket issues")
```
Comment (TusharKanekiDey) [inline test/test_utils/sagemaker.py:188]: I think @Satish615 had added this line to avoid Socket Failures.

Comment (tejaschumbalkar) [reply test/test_utils/sagemaker.py:188]: This line itself causes socket error as the sleep time is 300 sec

Comment (TusharKanekiDey) [reply test/test_utils/sagemaker.py:188]: Thanks for solving the issue.

**Pull request merged**